src/pages/workflows/workflow/WorkflowDetails.js

```diff
@@ -0,0 +1,304 @@
+  const signal = Utils.useCancellation()
+  const cachedSnapshotsList = Utils.useStore(snapshotsListStore)
+
+
+  const snapshotsList = cachedSnapshotsList && _.isEqual({ namespace, name }, _.pick(['namespace', 'name'], cachedSnapshotsList[0])) ?
+    cachedSnapshotsList :
+    undefined
+
+  Utils.useOnMount(() => {
+    const loadSnapshots = async () => {
+      snapshotsListStore.set(snapshotsList || await Ajax(signal).Methods.list({ namespace, name }))
+    }
+
+    if (!snapshotsList) {
+      loadSnapshots()
+    }
+  })
+
+
+  return h(Fragment, [
+    h(TopBar, { title: 'Workflows', href: Nav.getLink('workflows') }, [
+      div({ style: Style.breadcrumb.breadcrumb }, [
+        div(breadcrumbs.commonPaths.workflowList()),
+        div({ style: Style.breadcrumb.textUnderBreadcrumb }, [`${namespace}/${name}`])
+      ])
+    ]),
+    snapshotsList ?
+      children :
+      centeredSpinner()
+  ])
+}
+
+const SnapshotWrapper = ({ namespace, name, snapshotId, tabName, children }) => {
+  const signal = Utils.useCancellation()
+  const cachedSnapshotsList = Utils.useStore(snapshotsListStore)
+  const cachedSnapshot = Utils.useStore(snapshotStore)
+  const selectedSnapshot = (snapshotId * 1) || _.last(cachedSnapshotsList).snapshotId
+
+  const snapshot = cachedSnapshot &&
+  _.isEqual({ namespace, name, snapshotId: selectedSnapshot }, _.pick(['namespace', 'name', 'snapshotId'], cachedSnapshot)) ?
+    cachedSnapshot :
+    undefined
+
+  Utils.useOnMount(() => {
+    const loadSnapshot = async () => {
+      snapshotStore.set(await Ajax(signal).Methods.method(namespace, name, selectedSnapshot).get())
+    }
+
+    if (!snapshot) {
+      loadSnapshot()
+    }
+
+    if (!snapshotId) {
+      window.history.replaceState({}, '', Nav.getLink('workflow-dashboard', { namespace, name, snapshotId: selectedSnapshot }))
+    }
+  })
+
+  return h(Fragment, [
+    h(TabBar, {
+      activeTab: tabName,
+      tabNames: ['dashboard', 'wdl', 'configs'],
+      displayNames: { configs: 'configurations' },
+      getHref: currentTab => Nav.getLink(`workflow-${currentTab}`, { namespace, name, snapshotId: selectedSnapshot })
+    }, [
+      div({ style: { marginRight: '1rem' } }, ['Snapshot:']),
+      div({ style: { width: 100 } }, [
+        h(Select, {
+          value: selectedSnapshot,
+          isSearchable: false,
+          options: _.map('snapshotId', cachedSnapshotsList),
+          onChange: ({ value }) => Nav.goToPath(`workflow-${tabName}`, { namespace, name, snapshotId: value })
+        })
+      ])
+    ]),
+    snapshot ?
+      children :
+      centeredSpinner()
+  ])
+}
+
+const WorkflowSummary = () => {
+  const { namespace, name, snapshotId, createDate, managers, synopsis, documentation, public: isPublic } = Utils.useStore(snapshotStore)
+  const [importUrlCopied, setImportUrlCopied] = useState()
+  const importUrl = `${getConfig().orchestrationUrlRoot}/ga4gh/v1/tools/${namespace}:${name}/versions/${snapshotId}/plain-WDL/descriptor`
```

Comment:
Is this a new feature? I couldn't find this in FC UI. I'm guessing this is for composing WDLs?

Reply:
It's in the wdl tab in fc